src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -32,11 +35,11 @@ public synchronized Product getProduct(Long id) throws NoSuchElementException {
     @Override
     public synchronized List<Product> findProducts(String query, SortField sortField, SortOrder sortOrder) {
         if (query != null) {
-            ToIntFunction<Product> getNumberOfMatches = product -> (int) Arrays.stream(query.toLowerCase().split(" "))
-                    .filter(product.getDescription().toLowerCase()::contains)
-                    .count();
             if (sortField != null)
             {
+                ToIntFunction<Product> getNumberOfMatches = product -> (int) Arrays.stream(query.toLowerCase().split(" "))
+                        .filter(product.getDescription().toLowerCase()::contains)
+                        .count();
```

Comment:
Please extract repeating code into a method: Ctrl + Alt + M in IntelliJ IDEA.

Having duplicate code it becomes difficult to support it: suppose we need to change this getNumberOfMatches implementation - now we have to do this twice.